### ISSUE TITLE
BUG: `pt.stack` inputs type hint is incomplete

### ISSUE THREAD

**Issue by michaelosthege** [pymc-devs/pytensor#193]
### Describe the issue:

The function also accepts non-symbolic scalar types such as `np.number, float, int, builtins.complex` as well as scalar `Variable`s, but it's advertised as `Sequence[TensorVariable]`.

### Reproducable code example:

```python
pt.stack([1, 2])  # works
```

### Error message:

```shell
mypy reports things like

error: Argument 1 to "stack" has incompatible type "Sequence[Union[int, ndarray[Any, Any], Variable[Any, Any]]]"; expected "Sequence[TensorVariable[Any, Any]]"
```

### PyTensor version information:

2.9.1

### Context for the issue:

Detected by mypy job in https://github.com/pymc-devs/pymc/actions/runs/3903049911/jobs/6666888669 from https://github.com/pymc-devs/pymc/pull/6431

**Comment by covertg** [comment]
Would the preferred solution just be to update the input type to a Union of scalar and array types?

I suppose that a similar issue might exist for many pytensor functions that work with any of: Variable, scalar, and numpy types?

**Comment by michaelosthege** [comment]
yes, that's pretty much it

**Comment by michaelosthege** [comment]
Closed by #205
